reposerver/cache/cache.go

```diff
@@ -304,6 +304,28 @@ func (c *Cache) SetRevisionMetadata(repoURL, revision string, item *appv1.Revisi
 	return c.cache.SetItem(revisionMetadataKey(repoURL, revision), item, c.repoCacheExpiration, false)
 }
 
+func gitDataKey(gitObject, repoURL, revision string) string {
+	return fmt.Sprintf("%s|%s|%s", gitObject, repoURL, revision)
+}
+
+func (c *Cache) SetGitFiles(repoURL, revision string, files map[string][]byte) error {
+	return c.cache.SetItem(gitDataKey("gitfiles", repoURL, revision), &files, c.repoCacheExpiration, false)
```

Comment:
We create global constants for storing the keys `gitfiles` and `gitdirs`. That way we do not have to update 10 different places in case we decide to change the key.